cmd/otel-allocator/main.go

```diff
@@ -209,6 +212,21 @@ func main() {
 			setupLog.Info("Closing watcher loop")
 			close(eventCloser)
 		})
+	runGroup.Add(
+		func() error {
+			for {
+				select {
+				case scrapeCfg := <-scrapeConfigChan:
+					srv.UpdateScrapeConfigResponse(scrapeCfg)
```

Comment:
One thing I'm wondering is how to handle update failures such as unmarshalling. For now I tried to keep the behavior we have currently (return server error in the HTTP handler), but I wonder if we should not handle it here with `setupLog.Error` and keep returning last known good scrape configuration? Thoughts?

Reply:
No longer an open question I guess